libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2325,6 +2335,8 @@ bool TypeChecker::visit(FunctionCallOptions const& _functionCallOptions)
 
 	_functionCallOptions.expression().accept(*this);
 
+	_functionCallOptions.annotation().isPure = *_functionCallOptions.expression().annotation().isPure;
```

Comment:
Was it like that before? What about `f{value: functionWithSideEffects()}()`?

Reply:
Hmm yeah fair point, I don't know why I didn't just set it to `false`